src/components/Auth/SignUp/SignUp.js

```diff
@@ -1,97 +1,95 @@
+      .required('Pole obowiązkowe')
+      .min(6, 'Minimum 6 znaków'),
+  password: Yup.string()
+      .min(6, 'Minimum 6 znaków')
+      .max(40, 'Maxymalnie 30 znaków')
+      .required('Pole obowiązkowe'),
+  confirmPassword: Yup.string()
+      .oneOf([Yup.ref('password'), null], 'Hasła muszą być identyczne')
+      .required('Pole obowiązkowe')
+});
 
-      await createUserProfileDocument(user, { displayName });
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
-  const handleChange = (event) => {
-    const { name, value } = event.target;
-
-    setUserCredentials({ ...userCredentials, [name]: value });
-  };
+const SignUp = () => {
 
   return (
-    <div className='sign-up'>
-      <h2>Marzy mi się założenie konta</h2>
-      <p>Rejestracja</p>
-      <form onSubmit={handleSubmit}>
-        <Input
-          name='displayName'
-          type='text'
-          label='Display Name'
-          value={displayName}
-          onChange={handleChange}
-          required
-        />
-        <Input
-          name='email'
-          type='email'
-          label='Email'
-          value={email}
-          onChange={handleChange}
-          required
-        />
-
-        <Input
-          name='password'
-          type='password'
-          label='Password'
-          value={password}
-          onChange={handleChange}
-          required
-        />
-        <Input
-          name='confirmPassword'
-          type='password'
-          label='Confirm Password '
-          value={confirmPassword}
-          onChange={handleChange}
-          required
-        />
-        <div className='sign-in__buttons'>
-          <Button value='submit'>Załóż konto</Button>
-        </div>
-      </form>
-    </div>
+      <div className='sign-up'>
+        <h2>Marzy mi się założenie konta</h2>
+        <p>Rejestracja</p>
+        <Formik
+            initialValues={{
+              email: '',
+              password: '',
+              confirmPassword: ''
+            }}
+            validationSchema={SignInSchema}
+            onSubmit={async (values, {setSubmitting}) => {
+              if (values.password !== values.confirmPassword) {
+                alert('hasła są niezgodne');
+                return;
+              }
+              try {
+                await auth.createUserWithEmailAndPassword(
+                    values.email,
+                    values.password
+                );
+              } catch (error) {
+                console.error(error);
+              }
+              setSubmitting(false);
+            }}
+        >
+          {({
+              values, handleChange,
+            }) => (
+              <Form>
+                <Field
+                    label="E-mail"
+                    id="emailId"
+                    name="email"
+                    type="email"
+                    component={Input}
+                    value={values.email}
+                    onChange={handleChange}
```

Comment:
to samo gdzie jest ta metoda nie rozumiem tego?

Reply:
w 42  lini kodu  ja 
            values, handleChange,